src/extension/serializer.ts

```diff
@@ -852,8 +862,187 @@ export class GrpcSerializer extends SerializerBase {
+    this.ready = new Promise((resolve) => {
+      resolve()
+    })
+  }
+
+  createSerializerClient = () => {
+    this.log.info(`Serializer: Client pointed to: ${this.serializerServiceUrl}`)
+    // Server options:
+    // (1) pre-existing, started internally by this extension (assuming local execution so it has permissions to start)
+    // (2) pre-existing, started externally, presumably at the this.serializerServiceUrl address
+    //   In both cases we need a key & cert (assuming tls) which may reside in:
+    //   (a) this project: '/path/to/projectRoot/tls'
+    //   (b) the runme binary server default: '~/.config/runme/tls'
+    //   (c) other, manually configured
+
+    // I will assume 2a here
+    const tlsPath = path.normalize(__dirname + '/../tls')
+    const keyPath = `${tlsPath}/key.pem`
+    const certPath = `${tlsPath}/cert.pem`
+
+    let grpcOptions = {
+      baseUrl: this.serializerServiceUrl,
+      httpVersion: '2',
+      nodeOptions: {
+        rejectUnauthorized: false, // allow self-signed certificates
+        cert: fs.readFileSync(certPath),
+        key: fs.readFileSync(keyPath),
+      },
+    } as GrpcTransportOptions
+
+    return createPromiseClient(ParserService, createGrpcTransport(grpcOptions))
+
+    // return createConnectTransport({
+    //   baseUrl: this.serializerServiceUrl,
+    //   httpVersion: '1.1',
+    // })
+  }
+
+  protected async saveNotebook(
+    data: NotebookData,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    token: CancellationToken,
+  ): Promise<Uint8Array> {
+    const notebook = Marshal.notebook(data, 'es_proto') as es_proto.Notebook
+    const request = new es_proto.SerializeRequest({ notebook })
+    const response = await this.client.serialize(request)
+    return response.result
+  }
+
+  protected async reviveNotebook(
+    content: Uint8Array,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    token: CancellationToken,
+  ): Promise<Serializer.Notebook> {
+    // const markdown = Buffer.from(content).toString('utf8')
+    const deserializeRequest = new es_proto.DeserializeRequest()
+    deserializeRequest.source = content
+    let res
+    try {
+      res = await this.client.deserialize(deserializeRequest)
+    } catch (e: any) {
+      if (e.name === 'ConnectError') {
+        e.message = `Unable to connect to the serializer service at ${this.serializerServiceUrl}`
+      }
+      log.error('Error in reviveNotebook  ', e as any)
+      throw e
+    }
+    const notebook = res.notebook
+
+    if (!notebook) {
+      return this.printCell('⚠️ __Error__: no cells found!')
+    }
+    return notebook as any as Serializer.Notebook // ugly cast :(
+  }
+
+  protected async saveNotebookOutputsByCacheId(_cacheId: string): Promise<number> {
+    console.error('saveNotebookOutputsByCacheId not implemented for WasmSerializer')
+    return -1
+  }
+
+  public async saveNotebookOutputs(_uri: Uri): Promise<number> {
```

Comment:
When I refer to caching I mean the caching done via `saveNotebookOutputsByCacheId` which is what powers the `saveNotebookOutputs` method.

Reply:
I will be looking at resolving this tomorrow